### PR TITLE
Updated to latest chart.js, ESM build, types and updated test

### PR DESCRIPTION
**Pull request by santam85**
Thanks for contributing.

- [X] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

Updated Chart.js peer dependency to the latest version, and fixed broken tests.
Added ESM build to avoid warnings when using the plugin with Angular (https://github.com/valor-software/ng2-charts/pull/1276)

### CODE REVIEW

File: src/controller.financial.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿'use strict';
 
-import {BarController} from 'chart.js';
+import {BarController, Chart} from 'chart.js';
```
Comment (benmccann) [inline src/controller.financial.js:3]: maybe this should be `defaults` instead of `Chart` and then down below change `Chart.defaults` to just `defaults`

Comment (benmccann) [reply src/controller.financial.js:3]: Thanks for raising this as an issue. I just pushed a change to `master` to fix this part

File: package-lock.json [hunk 2]
```diff
@@ -1,3883 +1,8 @@
 {
   "name": "chartjs-chart-financial",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (benmccann) [inline package-lock.json:4]: I don't know that we want to go back and forth between lockfile versions with every commit. Is it possible fo you to upgrade to a newer version of `npm`?

Comment (santam85) [reply package-lock.json]: Sorry didn't realise. Done.

File: test/specs/controller.financial.tests.js [hunk 3]
```diff
@@ -1,28 +1,22 @@
...
-						{t: 1491091200, o: 60.23, h: 200.23, l: 50.29, c: 90.20},
-						{t: 1491177600, o: 40.00, h: 50.42, l: 20.31, c: 30.20},
-						{t: 1491264000, o: 60.90, h: 80.34, l: 40.14, c: 60.29},
-						{t: 1491350400, o: 60.94, h: 100.09, l: 50.78, c: 90.04},
-						{t: 1491436800, o: 30.49, h: 50.20, l: 20.09, c: 40.20},
-						{t: 1491523200, o: 50.04, h: 80.93, l: 40.94, c: 70.24},
-						{t: 1491609600, o: 60.29, h: 100.49, l: 50.49, c: 90.24},
-						{t: 1491696000, o: 40.04, h: 50.23, l: 20.23, c: 30.49},
-						{t: 1491782400, o: 30.23, h: 50.09, l: 20.87, c: 40.49}
+						{x: date, t: 1491004800, o: 70.87, h: 80.13, l: 40.94, c: 50.24},
```
Comment (benmccann) [inline test/specs/controller.financial.tests.js:8]: Hmm. We don't want to have to specify `x` and `t`. And `x` shouldn't be the same `date` everywhere. 

Ideally we could change the scale ID to `t` and have that work still. We may have to set the `position` to `'bottom'` or `axis` to `'x'`: https://www.chartjs.org/docs/latest/axes/cartesian/.  Though I'm not quite sure that'd be enough. I think the bar chart may look for the parameter to be called `x`. I just changed the samples to `x` in https://github.com/chartjs/chartjs-chart-financial/commit/0b4555efbfb82a7b665ec4bda475c17e17267553 but am not sure I'm really happy with that.

Or we could turn `parsing` on to make `t` work: https://www.chartjs.org/docs/latest/general/data-structures.html. But that would have some performance impact

I haven't really thought through what the best solution is here

Comment (santam85) [reply test/specs/controller.financial.tests.js]: Took the most naive approach and matched the sample for the test as well, using x for the time axis. Checking other plugins and the default examples, it seemed to make the most sense for consistency.

**Comment by santam85** [comment]
@benmccann can you have another look at the latest iteration of this? As things are standing now, the master branch would not even build for me on the latest Chart.js with the following error:
```
Error: ./node_modules/chartjs-chart-financial/src/controller.candlestick.js 47:3-8
"export 'default' (imported as 'Chart') was not found in 'chart.js'
````

**Review by PowerKiKi — APPROVED** [review]

**Comment by benmccann** [comment]
This PR is getting quite difficult to review as more changes are tacked on to it. I think most of the changes are good, but it's a bit hard for me to even tell what's changed especially with the indentation changed. Would you be able to split this up into separate PRs?

**Comment by santam85** [comment]
@benmccann enforced tab indentation in eslint so that those changes are not confusing you and won't be added inadvertently in the future. Unfortunately it's quite difficult to split these changes up in a way they are self consistent: the changes required for ESM are pervasive.

**Comment by benmccann** [comment]
I do think it could be split up a bit. I just pushed two of the changes separately. If you rebase now hopefully it should make this PR at least a bit smaller

**Comment by santam85** [comment]
@benmccann any new comments? Do you want me to update the README and the docs too?

**Comment by Sir-J** [comment]
+1

**Comment by Eugeno** [comment]
Are there any updates?

**Comment by santam85** [comment]
@benmccann I'd like some closure on this PR in order to understand if we need to remove the financial charts plugin from ng2-charts. If you or any other maintainer could give a final review to the PR and tell us what needs to be addressed, I'd be very grateful. Otherwise please reject and close.

**Comment by dostarora97** [comment]
@benmccann Any updates on this PR ?
I believe quite a lot of people depend on this PR, since its' incorporated in ng-charts.
Could you plz provide us any updates ?

**Comment by amer-yousuf** [comment]
@benmccann any update on this PR, please?

**Comment by santam85** [comment]
@etimberg @kurkle Perhaps you can be of help here?

File: src/index.esm.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (kurkle) [inline src/index.esm.js:1]: nit: I think this could be omitted

**Review by kurkle — APPROVED** [review]
I saw no blockers.
I know @benmccann would like the `t` key work out of the box for `x` axis, but that is yet another change and should not be included in this PR.

**Comment by kurkle** [comment]
Thanks for the patience, @santam85 
I think it will take at least couple of weeks before @etimberg is available, no idea how busy @benmccann is nowadays. I would not like to step on any toes by just merging this.

**Comment by benmccann** [comment]
This has been on my queue for awhile. Sorry I haven't gotten to it yet. I will take a look shortly

File: karma.conf.js [hunk 5]
```diff
@@ -49,17 +48,17 @@ module.exports = function(karma) {
...
 		rollupPreprocessor: {
-			plugins: [
-				nodeResolve(),
-				commonjs()
-			],
 			external: [
-				'chart.js'
+				'chart.js',
+				'chart.js/helpers',
+				'chartjs-chart-financial',
```
Comment (benmccann) [inline karma.conf.js:54]: I'm not a karma expert. Can you add a comment explaining why this need to be `external` / global?

File: rollup.config.js [hunk 6]
```diff
@@ -17,47 +15,48 @@ const banner = `/*!
...
+					'chart.js': 'Chart',
+					'chart.js/helpers': 'Chart.helpers',
+				},
+			};
+
+			if (suffix.match(/\.min\.js$/)) {
+				config.plugins.push(
+					terser({
+						output: {
+							comments: /^!/,
```
Comment (benmccann) [inline rollup.config.js:36]: can you add a comment explaining what this does? I looked briefly at the terser docs and think it keeps comments beginning with `!`? which comments would those be?

**Comment by benmccann** [comment]
thank you for this! this is a really nice set of changes and looks good to me as well. I just left two minor comments along with the one @kurkle left. if you don't mind making those minor updates we can go ahead and merge this

sorry for being MIA on this. I kept meaning to get back to it. thanks for being persistent while being patient and understanding at the same time

File: karma.conf.js (hunk 5 above)
Comment (santam85) [reply karma.conf.js:54]: It's not directly a Karma option, it comes from the rollupPreprocessor: https://rollupjs.org/guide/en/#external
AFAIU, it has to be done this way to ensure the library is not included in the karma bundle and the tests run using the UMD version of the built files, similar to how end-users would use the library if not relying on a builder.
I adopted the same pattern found in others mainstream chart.js plugins.

File: rollup.config.js (hunk 6 above)
Comment (santam85) [reply rollup.config.js:36]: Those would generally be the license comment blocks, including dependencies. Normally we are required to include those in the built output.

Comment (kurkle) [reply rollup.config.js:36]: I don't think this option makes any change to the output though, so maybe its not needed.

From terser docs:

> By default this works like Google Closure, keeping JSDoc-style comments that contain "@license" or "@preserve".

**Comment by santam85** [comment]
> thank you for this! this is a really nice set of changes and looks good to me as well. I just left two minor comments along with the one @kurkle left. if you don't mind making those minor updates we can go ahead and merge this
> 
> sorry for being MIA on this. I kept meaning to get back to it. thanks for being persistent while being patient and understanding at the same time

No problem, we're all busy with our day jobs 😄 
Applied the new suggestions, let me know if there's more.

**Review by kurkle — APPROVED** [review]

**Comment by santam85** [comment]
fixes #99

**Pull request merged**